lib/fswatcher/fswatcher.go

```diff
@@ -0,0 +1,192 @@
+	for {
+		watcher.resetNotifyTimerIfNeeded()
+		select {
+		case event, _ := <- watcher.fsEventChan:
+			watcher.speedUpNotifyTimer()
+			watcher.storeFsEvent(event)
+		case <-watcher.notifyTimer.C:
+			watcher.sendStoredEventsToModelOrSlowDownTimer()
+		case event := <-finishedFileEventSubscription.C():
+			watcher.skipPathChangedByUs(event)
+		}
+	}
+}
+
+func (watcher *FsWatcher) newFsEvent(eventPath string) *FsEvent {
+	if isSubpath(eventPath, watcher.folderPath) {
+		path := relativePath(eventPath, watcher.folderPath)
+		if !isSpecialPath(path) {
+			return &FsEvent{path}
+		}
+	}
+	return nil
+}
+
+func relativePath(path string, folderPath string) string {
+	subPath, _ := filepath.Rel(folderPath, path)
+	return subPath
+}
+
+func isSubpath(path string, folderPath string) bool {
+	if len(path) > 1 && os.IsPathSeparator(path[len(path) - 1]) {
+		path = path[0:len(path)-1]
+	}
+	if len(folderPath) > 1 && os.IsPathSeparator(folderPath[len(folderPath) - 1]) {
+		folderPath = folderPath[0:len(folderPath)-1]
+	}
+	return strings.HasPrefix(path, folderPath)
+}
+
+func (watcher *FsWatcher) resetNotifyTimerIfNeeded() {
+	if watcher.notifyTimerNeedsReset {
+	l.Debugf("Resetting notifyTimer to %#v\n", watcher.notifyDelay)
+		watcher.notifyTimer.Reset(watcher.notifyDelay)
+		watcher.notifyTimerNeedsReset = false
+	}
+}
+
+func (watcher *FsWatcher) speedUpNotifyTimer() {
+	if watcher.notifyDelay != fastNotifyDelay {
+		watcher.notifyDelay = fastNotifyDelay
+		l.Debugf("Speeding up notifyTimer to %#v\n", watcher.notifyDelay)
+		watcher.notifyTimerNeedsReset = true
+	}
+}
+
+func (watcher *FsWatcher) slowDownNotifyTimer() {
+	if watcher.notifyDelay != slowNotifyDelay {
+		watcher.notifyDelay = slowNotifyDelay
+		l.Debugf("Slowing down notifyTimer to %#v\n", watcher.notifyDelay)
+		watcher.notifyTimerNeedsReset = true
+	}
+}
+
+const (
+	slowNotifyDelay = time.Duration(60) * time.Second
+	fastNotifyDelay = time.Duration(500) * time.Millisecond
+)
+
+func (watcher *FsWatcher) addEvent(event FsEvent) {
+	watcher.fsEvents[event.path] = event
+}
+
+func (watcher *FsWatcher) removeEventIfPresent(path string) {
+	delete(watcher.fsEvents, path)
+}
+
+func (watcher *FsWatcher) storeFsEvent(event notify.EventInfo) {
+	newEvent := watcher.newFsEvent(event.Path())
+	if newEvent != nil {
+		watcher.addEvent(*newEvent)
+	}
+}
+func (watcher *FsWatcher) sendStoredEventsToModelOrSlowDownTimer() {
+	watcher.notifyTimerNeedsReset = true
+	if len(watcher.fsEvents) > 0 {
+		l.Debugf("Notifying about %d fs events\n", len(watcher.fsEvents))
+		watcher.notifyModelChan <- watcher.fsEvents
+	} else {
+		watcher.slowDownNotifyTimer()
+	}
+	watcher.fsEvents = make(map[string]FsEvent)
+}
+
+func (watcher *FsWatcher) skipPathChangedByUs(event events.Event) {
+	path := event.Data.(map[string]interface{})["item"].(string)
+	l.Debugf("Skipping notification for finished path: %s\n", path)
+	watcher.removeEventIfPresent(path)
```

Comment:
I think you need to be a bit more clever here. 
Have a list of files you are currently ignoring, and add them to the list upon events.ItermStarted and remove them upon events.ItemFinished.